src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -816,7 +816,7 @@ private void appendToWAL() {
 
   private void beginAppend() {
     if (!appended.contains(WAL.beginMarker)) {
-      wal.append(WAL.beginMarker, "");
+      wal.append(WAL.beginMarker, ""); //this creates a cache entry
```

Comment:
`wal.append` in the `FSWAL.java` class calls two methods on the writer
```
      writer.append(key, value);
      writer.hsync();
```
Does this create two cache entries? above you mention a single one. If two entries are created and only one is cleaned that could also mean accumulation over time.

Reply:
@dosvath calling `writer.append(key, value);` and `writer.hsync();` doesn't create 2 cache entries(the size of `org.apache.hadoop.fs.FileSystem.Cache#map` doesn't change when I step over these lines in debugger).
Calling `wal.append(WAL.beginMarker, "");` creates one cache entry, this is a stack trace that I was able to get while running `HdfsSinkTaskTest`:
```
getInternal:3310, FileSystem$Cache (org.apache.hadoop.fs)
getUnique:3270, FileSystem$Cache (org.apache.hadoop.fs)
newInstance:528, FileSystem (org.apache.hadoop.fs)
<init>:146, WALFile$Writer (io.confluent.connect.hdfs.wal)
createWriter:78, WALFile (io.confluent.connect.hdfs.wal)
acquireLease:74, FSWAL (io.confluent.connect.hdfs.wal) --> we create a writer here
append:57, FSWAL (io.confluent.connect.hdfs.wal)
createWALs:365, HdfsSinkTaskTest (io.confluent.connect.hdfs)
testSinkTaskStartWithRecovery:233, HdfsSinkTaskTest (io.confluent.connect.hdfs)
...
```
So, essentially, creating one writer creates a one cache entry, and leaving unclosed writers(e.g. we catch and re-throw and exception in `io.confluent.connect.hdfs.wal.FSWAL#acquireLease`) blows up the cache.